pygit2/repository.py

```diff
@@ -1549,8 +1550,42 @@ def get(self, key: str):
             return None
 
     def __iter__(self):
-        for ref_name in self._repository.listall_references():
-            yield ref_name
+        iter = self._repository.references_iterator_init()
+        while True:
+            ref = self._repository.references_iterator_next(iter)
+            if ref:
+                yield ref.name
+            else:
+                return
```

Comment:
I think this could be written as `yield from self.iterator()`
Or even simpler, make them the same function `__iter__ = iterator`

Reply:
Forget that one, I read it too fast.